### PR TITLE
[interp] register callbacks for interpreter functions

### PR DESCRIPTION
**Pull request by lewurm**
first step to decouple interp from mini. the goal is to have the interpreter as a library available.

### CODE REVIEW

File: mono/mini/interp-stubs.c [hunk 1]
```diff
@@ -0,0 +1,180 @@
+#include <config.h>
```
Comment (vargaz) [inline mono/mini/interp-stubs.c]: Why was this moved to mini/ ?

File: mono/mini/interp-stubs.c [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+#include "interp/interp.h"
+
+/* interpreter callback stubs */
+
+static MonoJitInfo*
+mono_interp_stub_find_jit_info (MonoDomain *domain, MonoMethod *method)
+{
```
Comment (vargaz) [inline mono/mini/interp-stubs.c]: Static functions don't need a mono_ prefix.

File: mono/mini/mini.h [hunk 3]
```diff
@@ -2363,6 +2363,53 @@ mono_bb_last_inst (MonoBasicBlock *bb, int filter)
...
+	MonoJitInfo* (*frame_get_jit_info) (MonoInterpFrameHandle frame);
+	gpointer (*frame_get_ip) (MonoInterpFrameHandle frame);
+	gpointer (*frame_get_arg) (MonoInterpFrameHandle frame, int pos);
+	gpointer (*frame_get_local) (MonoInterpFrameHandle frame, int pos);
+	gpointer (*frame_get_this) (MonoInterpFrameHandle frame);
+	MonoInterpFrameHandle (*frame_get_parent) (MonoInterpFrameHandle frame);
+	void (*start_single_stepping) (void);
+	void (*stop_single_stepping) (void);
+} MonoInterpCallbacks;
+
```
Comment (vargaz) [inline mono/mini/mini.h]: This could go to interp.h with a forward declaration, i.e.:
struct _MonoInterpCallbacks;

typedef struct _MonoInterpCallbacks MonoInterpCallbacks;

and in interp.h:

struct _MonoInterpCallbacks {

File: mono/mini/interp-stubs.c (hunk 1 above)
Comment (lewurm) [reply mono/mini/interp-stubs.c]: I want everything out of the `interp/` directory that should not end up in the library. In order to avoid ifdef hell, the interp-stubs are now always part of the runtime, so there's a fallback for the callbacks available.

File: mono/mini/mini.h (hunk 3 above)
Comment (lewurm) [reply mono/mini/mini.h]: good point! I changed that and rebased this PR; I had to do a couple changes due to https://github.com/mono/mono/pull/6072 being merged. please review again 🙂

**Comment by kumpera** [comment]
Do we expect the interp vtable to ever change?
If we do, it must be versioned to reduce the chance of a build mistake installing a stale interp lib and hell breaking loose without us knowing why.
